src/core/config/resolver.rs

```diff
@@ -53,3 +56,87 @@ impl Resolver {
         }
     }
 }
+
+#[derive(Default, Clone, Debug, PartialEq, Eq, schemars::JsonSchema)]
+pub struct Resolvers(pub Vec<Resolver>);
```

Comment:
```suggestion
pub struct ResolverSet(pub Vec<Resolver>);
```

I think we can use the untagged feature of serde and convert this into an enum.

```rust
enum ResolverSet {
    One(Resolver)
    Many(Vec<Resolver>)
}
```

Reply:
that's what I've tried in the first place and the current implementation it's what I ended up, since we want:
- provide compatibility with previous definition where Resolver was flatten
- provide new option `resolvers: []` where it's not flatten anymore

There are limitations in serde to express this